thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/task/TaskInfoFactory.java

```diff
@@ -34,7 +35,8 @@ public static TaskInfo getTaskInfoFromTaskType(TaskType taskType, String taskInf
         case MONITOR:
           taskInfo = OBJECT_MAPPER.readValue(taskInfoString, MonitorTaskInfo.class);
           break;
-        case REPORTER:
+        case ALERT:
+          taskInfo = OBJECT_MAPPER.readValue(taskInfoString, AlertTaskInfo.class);
```

Comment:
once all components are complete, it may make more sense to use the switch case to define the class being deserialized, rather than defining the deserialization class and reading the value within each case. Alternatively you can do that now and have a null check for merge + default case.